projects/wave-core-new/src/lib/time/time-slider/time-slider.component.ts

```diff
@@ -0,0 +1,282 @@
+        private changeDetectorRef: ChangeDetectorRef,
+    ) {
+        this.subscriptions.push(
+            this.projectService.getLayerStream().subscribe((layerList) => {
+                if (layerList !== this.layerList) {
+                    this.layerList = layerList;
+                }
+                this.changeDetectorRef.markForCheck();
+            }),
+        );
+    }
+
+    ngOnInit(): void {
+        this.getTimelineData();
+        this.getTimelineGroups();
+        this.getOptions();
+
+        this.timeline = new Timeline(this.timelineContainer.nativeElement, this.data, this.options);
+        this.timeline.setGroups(this.groups);
+        this.timeline.setItems(this.data);
+
+        this.timeline.addCustomTime(this.startTime, 'start');
+        this.timeline.addCustomTime(this.endTime, 'end');
+
+        //listens to events of the custom timebars
+        this.timeline.on('timechanged', (properties) => {
+            if (properties.id === 'start') {
+                this.startTime = this.timeline?.getCustomTime(properties.id) as Date;
+            }
+            if (properties.id === 'end') {
+                this.endTime = this.timeline?.getCustomTime(properties.id) as Date;
+            }
+            if (!this.isRange) {
+                this.endTime = this.startTime;
+            }
+            this.changeTime();
+        });
+
+        this.timeline.on('rangechanged', (properties) => {
+            const startWindow = properties.start;
+            const endWindow = properties.end;
+            const windowSize = endWindow - startWindow;
+            this.changeSelectedScale(windowSize);
+        });
+
+        //Connection to the timestream of the project
+        //changes the position of the Timebars according to the Timestream of the project
+        this.subscriptions.push(
+            this.projectService.getTimeStream().subscribe((t) => {
+                this.startTime = t.start.toDate();
+                this.endTime = t.end.toDate();
+                this.changeDetectorRef.markForCheck();
+                if (this.isRange) {
+                    this.timeline?.removeCustomTime('end');
+                    this.isRange = false;
+                }
+                if (t.end.isAfter(t.start)) {
+                    this.timeline?.addCustomTime(t.end.toDate(), 'end');
+                    this.isRange = true;
+                }
+                this.timeline?.setCustomTime(t.start.toDate(), 'start');
+            }),
+        );
+    }
+
+    changeSelectedScale(windowSize: number): void {
+        this.changeDetectorRef.detectChanges();
+        //these scales have to be adjusted, when the component is moved somewhere else with a higher width
+        const yearScale = 1000 * 60 * 60 * 24 * 31 * 22; //22 months in milliseconds
+        const monthScale = 1000 * 60 * 60 * 24 * 15; //15 days in milliseconds
+        const dayScale = 1000 * 60 * 60 * 31; //31 hours in milliseconds
```

Comment:
Can't you calculate some heuristic? It looks like this is tailor-fit for your monitor then?

Reply:
Maybe its better like that 43acdcf7a771f785ccf6075c7bc923a94d22f3a8